compiler/rustc_middle/src/ty/context.rs

```diff
@@ -2224,6 +2224,20 @@ impl<'tcx> TyCtxt<'tcx> {
         })
     }
 
+    /// Given a `ty`, return whether it's an `impl Future<...>`.
+    pub fn ty_is_opaque_future(self, ty: Ty<'_>) -> bool {
```

Comment:
I'm pretty sure this can be used in multiple places! I recall seeing similar needs elsewhere.

Reply:
Hmm, this is the only other place I see looking at opaque Future types:
https://github.com/rust-lang/rust/blob/75337775f776250a3a29c951344186e698c11c75/compiler/rustc_infer/src/infer/error_reporting/mod.rs#L1834
that needs the `Output` type from the future - maybe I shouldn't have added this method and I should just use `get_impl_future_output_ty().is_some()` instead?

Reply:
Yes, that'd be a good idea. Could you make a small PR for it at some point?